packages/browser/src/login/oidc/ClientRegistrar.spec.ts

```diff
@@ -193,6 +195,94 @@ describe("ClientRegistrar", () => {
       );
       expect(client.clientId).toBe("an id");
       expect(client.clientSecret).toBe("a secret");
+      expect(client.clientExpiresAt).toBe(0);
+    });
+
+    it("registers a new client if the one in storage has expired", async () => {
+      const clientRegistrar = getClientRegistrar({
+        storage: mockStorageUtility(
+          {
+            "solidClientAuthenticationUser:mySession": {
+              clientId: "an id",
+              clientSecret: "a secret",
+              // The stored expiry is in the past:
+              clientExpiresAt: "5",
+            },
+          },
+          false
+        ),
+      });
+
+      const mockFetch = (jest.fn() as any).mockResolvedValueOnce(
+        /* eslint-disable camelcase */
+        new NodeResponse(
+          JSON.stringify({
+            client_id: "abcd",
+            client_secret: "1234",
+            client_secret_expires_at: 0,
+            redirect_uris: ["https://example.com"],
+          })
+        ) as unknown as Response
+        /* eslint-enable camelcase */
+      );
+      global.fetch = mockFetch;
+
+      await clientRegistrar.getClient(
+        {
+          sessionId: "mySession",
+          redirectUrl: "https://example.com",
+        },
+        {
+          ...IssuerConfigFetcherFetchConfigResponse,
+          registrationEndpoint: "https://some.issuer/register",
+        }
+      );
+
+      expect(mockFetch).toHaveBeenCalled();
+    });
+
+    it("registers a new client if the one in storage has no expiry stored", async () => {
```

Comment:
@NSeydoux I think this is what we'll want, because otherwise someone may have a valid client, but we hadn't stored the expiry, in which case registering a new client resolves that missing expiry.